assignment3.md

```diff
@@ -0,0 +1,233 @@
+- What if you include additional types of features, like paraphrases in the [Paraphrase Database](http://www.paraphrase.org) or the part-of-speech of context words?
+
+The only feature types that are off-limits are WordNet features.
+
+Turn in the predicted clusters that your VSM generates. Also provide a brief description of your method in a file called `writeup.pdf`, making sure to describe the vector space model you chose, the clustering algorithm you used, and the results of any preliminary experiments you might have run on the training set.
+
+The output file should be called: `test_clusters_features.txt`
+
+### 2. Questions
+
+Now that you have seen what a feature-based model can do, let's start working with dense word embeddings. We have provided a file called `question2.txt` that lists a few initial questions to get you started working with embeddings. We've also copied the initial questions below. Write your answers to each question on the specified line of `question2.txt` to turn in.
+
+- What is the dimensionality of these vectors?
+- What are the 5 most similar words to `picnic`?
+- Which of these words is not like the others: "
+- Solve the following analogy: "leg" is to "jump" as X is to "throw"
+
+### 3. Dense Representations
+Finally, we'd like to see if dense word embeddings are better for clustering the words in our test set. Run the word clustering task again, but this time use a dense word representation. 
+
+For this task, use files:
+* `/home1/m/mayhew/data/wiki-news-300d-1M.vec.bin`: fasttext word vectors (binary format, 300 dimensions)
+* `/home1/m/mayhew/data/GoogleNews-vectors-negative300.bin`: Google word2vec vectors (binary format, 300 dimensions)
+* `question3.py`: to load dense vectors and cluster. (this will be similar to `question1.py`)
+
+The baseline system for this section uses the provided word vectors to represent words, and K-means for clustering. 
+
+As before, achieving the baseline score will get you a B, but you might try to see if you can do better. Here are some ideas:
+- Train your own word vectors, either on the provided corpus or something you find online. Try experimenting with the dimensionality.
+- [Retrofitting](https://www.cs.cmu.edu/~hovy/papers/15HLT-retrofitting-word-vectors.pdf) is a simple way to add additional semantic knowledge to pre-trained vectors. The retrofitting code is available [here](https://github.com/mfaruqui/retrofitting). Experiment with different lexicons, or even try [counter-fitting](http://www.aclweb.org/anthology/N16-1018).
+
+As in question 1, turn in the predicted clusters that your dense vector representation generates. Also provide a brief description of your method in `writeup.pdf` that includes the vectors you used, and any experimental results you have from running your model on the training set.
```

Comment:
They should also do an analysis of different errors made by each system -- i.e. look at instances that the word-context matrix representation gets wrong and dense gets right, and vice versa, and see if there are any interesting patterns.

Reply:
updated.